src/firefly/js/visualize/draw/ShapeDataObj.js

```diff
@@ -268,19 +268,29 @@ const draw=  {
 
     getCenterPt(drawObj) {
         const {pts}= drawObj;
-        let xSum = 0;
-        let ySum = 0;
-        let xTot = 0;
-        let yTot = 0;
-
-        pts.forEach((wp) => {
-           xSum += wp.x;
-           ySum += wp.y;
-           xTot++;
-           yTot++;
-        });
 
-        return makeWorldPt(xSum / xTot, ySum / yTot);
+        if (pts && pts.length > 0) {
+            let xSum = 0;
+            let ySum = 0;
+            let xTot = 0;
+            let yTot = 0;
+
+            pts.forEach((wp) => {
+                xSum += wp.x;
+                ySum += wp.y;
+                xTot++;
+                yTot++;
+            });
+
+            const type = pts[0].type;
+            const x = xSum/xTot;
+            const y = ySum/yTot;
+
+            return type === Point.W_PT ? makeWorldPt(x, y) : Object.assign(new SimplePt(x, y), {type});
```

Comment:
We are trying not to use `new` except in factory functions.  It there is not one for `SimplePt` why don't you make one that takes `x,y,type`. That way you don't have to use `Object.assign` here either which exposes how  a point is created.